ompi/communicator/comm_cid.c

```diff
@@ -858,16 +859,36 @@ static int ompi_comm_allreduce_pmix_reduce_complete (ompi_comm_request_t *reques
     opal_dss.unload(&sbuf, (void**)&info.data.bo.bytes, &info.data.bo.size);
     OBJ_DESTRUCT(&sbuf);
 
-    if (cid_context->send_first) {
-        (void)asprintf(&info.key, "%s:%s:send:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
-        (void)asprintf(&pdat.value.key, "%s:%s:recv:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
+    bytesWritten = asprintf(&info.key,
+                            cid_context->send_first ? "%s:%s:send:%d"
+                                                    : "%s:%s:recv:%d",
+                            cid_context->port_string,
+                            cid_context->pmix_tag,
+                            cid_context->iter);
+
+    if (bytesWritten == -1) {
+        fprintf (stderr, "writing info.key failed\n");
```

Comment:
We don't typically `fprintf()` from deep within the library.  Instead, we typically use `opal_output()` if something needs to unconditionally be written to stderr (because it will prefix the output with a label indicating the process it came from).  For errors like this, however, we typically will use `opal_output_verbose()`, which lets the error only be displayed conditionally (i.e., if the user has selected a high enough verbose value at run time), or just let the caller decide whether to display the error or not.